magento/magento2
# Google Sitemap refactoring - introduce item resolver

Refactoring to introduce injectable item resolver as way to add items to sitemap

### Description
- Introduce SitemapItemResolverInterface with concrete implementations for cms pages, products, categories and aggregate
- Introduce SitemapItemInterface as representation of single sitemap item
- Introduce SitemapConfigReaderInterface
- Introduce SitemapItemResolverConfigInterface
- Remove addSitemapItems as it was introduced in develop branch and wasn't used yet as public API
- Deprecate helper (suggested by @vrann). Getting data from configuration is part of introduced concrete implementations.

### Fixed Issues (if relevant)

1. magento/magento2#10045: Refactoring: Adding items to the Sitemap

### Manual testing scenarios

1. Go to Marketing > SEO > Sitemap
2. Add sitemap - provide path and filename
3. Click generate & save sitemap
4. Check if content is properly generated.

### Contribution checklist
